### PR TITLE
Fix an issue calling incorrect api

### PR DESCRIPTION
**Pull request by ehossack**
This commit fixes a typo in calling the incorrect restrictions api (v1 version)

### CODE REVIEW

**Comment by ehossack** [comment]
I'm surprised there's no tests for the `get_download_url_for_file*` apis.

I'm hoping this is grounds for a 1.8.1 release as it breaks the download url api.

Additionally, some guidance (outside of this PR, updating your own docs) on how to have all the available versions of python for nox would be appreciated. For example macOS no longer supports 3.6.x python versions.

**Review by ppolewicz — APPROVED** [review]

**Comment by ppolewicz** [comment]
I think `nox` used `pyenv` to install all the python versions it wanted - I didn't install anything by myself and a full suite runs well on my box (except 3.5?), so I think the comment which suggests that you need to install all python versions on your OS by yourself is not accurate.

It would normally be important enough for a 1.8.1 release, but I think we'll have a 1.9.0 early next week with all the sync changes and v2 support anyway.

@mlech-reef could you please confirm what the situation is with installing python versions for the full nox suite to run tests locally? (we shouldn't merge the comment in if it's inaccurate)

**Comment by ppolewicz** [comment]
(I had to approve the PR for the tests to run)

**Comment by ehossack** [comment]
> I think `nox` used `pyenv` to install all the python versions it wanted - I didn't install anything by myself and a full suite runs well on my box (except 3.5?), so I think the comment which suggests that you need to install all python versions on your OS by yourself is not accurate.
> 
> It would normally be important enough for a 1.8.1 release, but I think we'll have a 1.9.0 early next week with all the sync changes and v2 support anyway.
> 
> @mlech-reef could you please confirm what the situation is with installing python versions for the full nox suite to run tests locally? (we shouldn't merge the comment in if it's inaccurate)

Makes sense if the comment is innacurate. I have `pyenv` configured and running any of the `nox` commands failed, and I couldn't get nox to figure out how to pick up my installed python versions - hence the `"which python"` in there. 
Also python 3.5 and 3.6 fail to install on my system (macOS 11) due to some limitations, just to keep that in mind for tests/prs. For my own project I just run all the commands in a docker container...

**Comment by ppolewicz** [comment]
Maybe it's a bug on macos, lets see what @mlech-reef will have to say about it

**Comment by mlech-reef** [comment]
I don't think `nox` installs python on your own. It never happens for me :) It uses versions from `['3.5', '3.6', '3.7', '3.8', '3.9']` by default. That list could be overridden via `PYTHON_VERSIONS` environment variable (it is documented in the developer guide). If the version is not available on the PATH, it will fail, e.g.:
image

If you have properly configured `global` versions in `pyenv` like:
image
or `local`, e.g.:
image

everything should work fine.

**Comment by ehossack** [comment]
So the comment itself is correct. And CI provides a backup.

File: test/unit/v_all/test_api.py [hunk 1]
```diff
@@ -61,3 +61,19 @@ def test_get_bucket_by_id_v2(self):
         self.cache.save_bucket(Bucket(api=self.api, name='bucket_name', id_='bucket_id'))
         read_bucket = self.api.get_bucket_by_id('bucket_id')
         assert read_bucket.name == 'bucket_name'
+
+    @pytest.mark.apiver(from_ver=0)
```
Comment (mlech-reef) [inline test/unit/v_all/test_api.py:65]: It is redundant

File: test/unit/v_all/test_api.py [hunk 2]
```diff
@@ -61,3 +61,19 @@ def test_get_bucket_by_id_v2(self):
...
+    def test_get_download_url_for_file_name(self):
+        self._authorize_account()
+
+        download_url = self.api.get_download_url_for_file_name('bucket1', 'some-file.txt')
+
+        assert download_url == 'http://download.example.com/file/bucket1/some-file.txt'
+
+    @pytest.mark.apiver(from_ver=0)
```
Comment (mlech-reef) [inline test/unit/v_all/test_api.py:73]: It is redundant

**Review by mlech-reef — CHANGES_REQUESTED** [review]

File: test/unit/v_all/test_api.py (hunk 1 above)
Comment (ehossack) [reply test/unit/v_all/test_api.py:65]: Is consistency across the tests not better than having some tests that have the annotation and some not? Happy to change it but as someone coming in to the code I would wonder why some have and some don't

**Comment by ehossack** [comment]
Moved the tangential conversation around local setup to #255

File: test/unit/v_all/test_api.py (hunk 1 above)
Comment (mlech-reef) [reply test/unit/v_all/test_api.py]: Why marking all the tests with some defaults? The reason for having this mark is to be able to easily list/run only the tests that differ across apiver versions with e.g. `nox -s unit-3.9 -- --co -m apiver`

Comment (ppolewicz) [reply test/unit/v_all/test_api.py]: I agree that we should remove it. It's clear to me that the decorator should only be used on tests which should execute for a limited subset of apiver interface versions.

**Comment by ppolewicz** [comment]
ok, my mistake, I was under the impression that nox will install environments for itself

**Review by mlech-reef — APPROVED** [review]

File: test/unit/v_all/test_api.py (hunk 1 above)
Comment (ehossack) [reply test/unit/v_all/test_api.py]: Fair enough. I already updated the PR to remove it - but sounds like it was just my confusion :)

Comment (mlech-reef) [reply test/unit/v_all/test_api.py]: Already approved :)

**Review by ppolewicz — APPROVED** [review]

**Comment by ppolewicz** [comment]
Great contribution, added a test. I think it's not the first thing @ehossack contributed, right? Perhaps it's time for some more ambitious work we could do together?

**Comment by ehossack** [comment]
> Great contribution, added a test. I think it's not the first thing @ehossack contributed, right? Perhaps it's time for some more ambitious work we could do together?

For now I will have to stay with the smaller contributions to save my time!

**Comment by ehossack** [comment]
P.S. When can we expect 1.9.0 if no 1.8.1? 😛

**Comment by ppolewicz** [comment]
It's very close now. @mpnowacki-reef do we have anything left for 1.9.0? A couple new options for sync I think?

**Pull request merged**